distutils/dist.py

```diff
@@ -1027,22 +1027,22 @@ def has_pure_modules(self) -> bool:
         return len(self.packages or self.py_modules or []) > 0
 
     def has_ext_modules(self) -> bool:
-        return self.ext_modules and len(self.ext_modules) > 0
+        return bool(self.ext_modules and len(self.ext_modules) > 0)
```

Comment:
Agreed returning a proper bool makes sense; what do you think about these alternatives?

```suggestion
        return bool(self.ext_modules) and len(self.ext_modules) > 0
```


```suggestion
    @apply(bool)
    def has_ext_modules(self) -> bool
        return self.ext_modules and len(self.ext_modules) > 0
```

Reply:
And now that I look at these functions again, I realize they're way over-engineered. Collections already have boolean support.

```suggestion
        return bool(self.ext_modules)
```

Reply:
To answer your first comment:
1. I indiscriminately wrapped the entire expression. Which means I didn't have to call `bool` twice anywhere.
2. That `@apply` trick would work if the decorator is properly typed to forward its parameter return type. But you also have to change the declared return type of `` or type-checkers will correctly point out that you're returning a `list` instead of a `bool`.